lib/mapper.js

```diff
@@ -15,45 +14,40 @@ var internals = {
      * @param out the object where mapped data is written
      */
     map: function map(name, spec, eventData, out) {
-        var columns, values, i, len, column, value;
+        let columns, i, len, column, value, line;
+
+        line = {
+            event: name,
+            fields: [],
+            timestamp: null
+        };
 
         columns = Object.keys(spec);
-        values = [];
 
         for (i = 0, len = columns.length; i < len; i++) {
             column = columns[i];
             value = spec[column];
             value = (typeof value === 'function') ? value(eventData) : eventData[value];
 
-            switch (typeof value) {
-                case 'string':
-                case 'boolean':
-                case 'number':
-                    // ignore
-                    break;
-
-                case 'object':
-                    if (Array.isArray(value)) {
-                        value = String(value);
-                    } else if (value instanceof Date) {
-                        value = String(value);
-                    } else {
-                        value = Stringify(value);
+            if (column === 'time') {
+                line['timestamp'] = value;
+            } else {
+                if(typeof value === 'object') {
+                  Object.keys(value).forEach(function(k, v) {
+                    let field = formatField(column + k, v);
+                    if(field) {
+                        line['fields'].push(field);
```

Comment:
why different notation for referencing properties? wherever you set properties of `line` you use bracket notation, whereas elsewhere (constructor, `formatLine`, etc.) you use dot notation.